IoTHubBuddy/IoTViewer/AccountManager.cs

```diff
@@ -0,0 +1,362 @@
+            Debug.WriteLine("provider: " + provider.Id + " scope: " + scope + " client: " + client);
+            if (provider.Id == "https://login.microsoft.com")
+            {
+                request.Properties.Add("resource", "https://management.azure.com/");
+            }
+            WebTokenRequestResult result = await WebAuthenticationCoreManager.RequestTokenAsync(request);
+            if (result.ResponseStatus == WebTokenRequestStatus.Success)
+            {
+                WebAccount account = result.ResponseData[0].WebAccount;
+                StoreWebAccount(account);
+                var sure = result.ResponseData[0];
+                string token = result.ResponseData[0].Token;
+
+                Debug.WriteLine("token: " + token);
+                mp.NavigateToSubscription(token);
+
+            }
+            
+
+        }
+
+        /// <summary>
+        /// store web account in local storage
+        /// NOTE: this will likely change soon
+        /// </summary>
+        /// <param name="account"></param>
+        private static async void StoreWebAccount(WebAccount account)
+        {
+            ApplicationDataContainer accountsContainer = ApplicationData.Current.LocalSettings.Containers[AccountsContainer];
+
+            if (account.UserName != "")
+            {
+                accountsContainer.Containers[ProviderIdSubContainer].Values[account.UserName] = account.WebAccountProvider.Id;
+                accountsContainer.Containers[AuthoritySubContainer].Values[account.Id] = account.WebAccountProvider.Authority;
+                
+            }
+            else
+            {
+                accountsContainer.Containers[ProviderIdSubContainer].Values[account.Id] = account.WebAccountProvider.Id;
+                accountsContainer.Containers[AuthoritySubContainer].Values[account.Id] = account.WebAccountProvider.Authority;
+            }
+            ApplicationData.Current.LocalSettings.Values["CurrentUserProviderId"] = account.WebAccountProvider.Id;
+            ApplicationData.Current.LocalSettings.Values["CurrentUserId"] = account.Id;
+            ApplicationData.Current.LocalSettings.Values["CurrentAuthority"] = account.WebAccountProvider.Authority;
+
+        }
+        /// <summary>
+        /// Attempt to get token silently (without prompting user for credentials) from stored account
+        /// </summary>
+        /// <returns></returns>
+        public static async Task<string> GetTokenSilentlyAsync()
+        {
+            string providerId = ApplicationData.Current.LocalSettings.Values["CurrentUserProviderId"]?.ToString();
+            string accountId = ApplicationData.Current.LocalSettings.Values["CurrentUserId"]?.ToString();
+            string authority = ApplicationData.Current.LocalSettings.Values["CurrentAuthority"]?.ToString();
+
+            if (null == providerId || null == accountId)
+            {
+                return null;
```

Comment:
Avoid 'return null' -- this just makes callers life miserable ("billion dollar mistake")